services/transientsource/transientsource.go

```diff
@@ -0,0 +1,140 @@
+package transientsource
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/utils/pubsub"
+	"github.com/tidwall/gjson"
+)
+
+// Service provides services related to transient source ids
+type Service interface {
+
+	// SourceIdsSupplier provides an up-to-date supplier of transient source ids
+	SourceIdsSupplier() func() []string
+
+	// Apply performs transient source_id filtering logic against a single source_id.
+	// If it corresponds to a transient source_id, the result will be true, otherwise false.
+	Apply(sourceId string) bool
+
+	// ApplyJob performs transient source_id filtering logic against a single job.
+	// If the job corresponds to a transient source_id, the result will be true, otherwise false.
+	ApplyJob(job *jobsdb.JobT) bool
+
+	// ApplyParams performs transient source_id filtering logic against a job's parameters.
+	// If the parameters contain a transient source_id, the result will be true, otherwise false.
+	ApplyParams(params json.RawMessage) bool
+}
+
+// NewService creates a new service that updates its transient source ids while
+// backend configuration gets updated.
+func NewService(ctx context.Context, config backendconfig.BackendConfig) Service {
+	s := &service{
+		init: make(chan struct{}),
+	}
+	s.updateLoop(ctx, config)
+	return s
+}
+
+// NewEmptyService creates a new service that operates against an empty list of transient source ids
+// Useful for tests, when you are not interested in testing for transient sources.
+func NewEmptyService() Service {
+	return NewStaticService([]string{})
+}
+
+// NewStaticService creates a new service that operates against a predefined list of transient source ids.
+// Useful for tests.
+func NewStaticService(sourceIds []string) Service {
+	s := &service{
+		init:         make(chan struct{}),
+		sourceIds:    sourceIds,
+		sourceIdsMap: asMap(sourceIds),
+	}
+	close(s.init)
```

Comment:
[minor] Do we really need to make the channel and then close it? Can't we just not make it?